### ISSUE TITLE
Integrating wasm-pack into Game of Life

### ISSUE THREAD

**Issue by mgattozzi** [rustwasm/book#26]
In today's meeting we wanted to discuss how we would integrate wasm-pack into the Game of Life tutorial so we can teach people how to publish stuff as well. However, this makes things a bit difficult because how the tutorial is written to work and how wasm-pack works are a bit incompatible. This would require a bit of an undertaking to implement but what we need to discuss here is this:

- Do we want to do this or would it be better as an appendix resource as it currently is?
- If we do integrate it what would the scope be?
- How would we want to change the tutorial to work with it?
- How would we change the template to work with it?
- Are there other issues with this that we haven't discussed?

**Comment by mgattozzi** [comment]
cc/ @rustwasm/wasm-pack, @rustwasm/book, @rustwasm/core

**Comment by fitzgen** [comment]
> Do we want to do this or would it be better as an appendix resource as it currently is?

I think we want this integrated directly.

This is the blessed happy-path workflow we want to steer people towards, so our tutorial should show it to them.

> If we do integrate it what would the scope be?
>
> How would we want to change the tutorial to work with it?

The way I'm imagining it, we would have users publish their own GoL crate to npm. First, publishing the version that renders to a string, and then using that as an npm dependency with webpack in the webpage. Then, when implementing exposing the raw wasm memory for the cells to JS, we would have them publish a new version of the crate to npm, and then update the JS on the website.

Basically, make the code / workflow more like the existing wasm-pack tutorial, but with the GoL content.

> How would we change the template to work with it?

Perhaps we wouldn't use the existing template, but one of the new ones that @ag_dugs is working on?

**Comment by fitzgen** [comment]
I think this needs to be done for Rust 2018 edition, so adding the "Blocking Rust 2018" label.

**Comment by fitzgen** [comment]
PR for this: #78
